index.js

```diff
@@ -1,10 +1,20 @@
 'use strict';
 
-module.exports = function hasDynamicImport() {
-	try {
-		require('./import'); // eslint-disable-line global-require
-		return true;
-	} catch (e) {
-		return false;
+var util = require('util');
+
+var supported = false;
+
+try {
+	// eslint-disable-next-line global-require
+	var promise = require('./import');
+
+	// eslint-disable-next-line no-magic-numbers
+	if (util.inspect(promise).indexOf('<rejected>') === -1) {
```

Comment:
i don't think we should rely on node's magic `util.inspect` here, and either way, i think this is a race condition - there's no guarantee the promise will be settled by the time this line runs.

Reply:
i'm not so sure this is a race condition... i think `import()` returns a rejected promise when it is not supported. it seems to behave exactly like `Promise.reject()`.

without using logic like this i dont think it is possible to do this check synchronously, meaning this package needs to become async and return promises...

Reply:
Promise.reject is also not guaranteed to reject immediately.

I think that you’re right - which means we should leave the main entry point as is, and add a second entry point that returns a promise for a boolean. That way the async check only needs to be done when the syntax is actually supported.